iroh-net/src/magicsock/relay_actor.rs

```diff
@@ -1,5 +1,5 @@
 use std::{
-    collections::{BTreeMap, HashSet},
+    collections::{BTreeMap, BTreeSet},
```

Comment:
The aim is to save the cost of hashing.  Since these are just 32 bytes I think BTreeSet is very suitable, not sure fnv would be any better looking at the graphs linked from the fnv docs.  Maybe if the inputs were < 15 bytes it might be worth it to try fnv.